### ISSUE TITLE
Obsolete function warnings

### ISSUE THREAD

**Issue by Ergus** [emacsorphanage/gnuplot#61]
Hi:

When bytecompiling I am getting these warnings which are very simple to fix. 

```
Warning (bytecomp): Package cl is deprecated
Warning (bytecomp): ‘incf’ is an obsolete alias (as of 27.1); use ‘cl-incf’ instead.
Warning (bytecomp): ‘flet’ is an obsolete macro (as of 24.3); use either ‘cl-flet’ or ‘cl-letf’.
Warning (bytecomp): ‘case’ is an obsolete alias (as of 27.1); use ‘cl-case’ instead.
Warning (bytecomp): ‘destructuring-bind’ is an obsolete alias (as of 27.1); use ‘cl-destructuring-bind’ instead.
Warning (bytecomp): ‘case’ is an obsolete alias (as of 27.1); use ‘cl-case’ instead.
Warning (bytecomp): the following functions might not be defined at runtime:
    gnuplot-point-at-end-of-command,
    gnuplot-beginning-of-continuation, gnuplot-beginning-of-command,
    gnuplot-point-at-beginning-of-command,
    info-lookup-interactive-arguments, gnuplot--adjust-info-display
Warning (bytecomp): ‘member*’ is an obsolete function (as of 27.1); use ‘cl-member’ instead. [2 times]
Warning (bytecomp): ‘remove*’ is an obsolete function (as of 27.1); use ‘cl-remove’ instead. [2 times]
Warning (bytecomp): ‘member*’ is an obsolete function (as of 27.1); use ‘cl-member’ instead.
Warning (bytecomp): ‘remove*’ is an obsolete function (as of 27.1); use ‘cl-remove’ instead. [10 times]
Warning (bytecomp): ‘flet’ is an obsolete macro (as of 24.3); use either ‘cl-flet’ or ‘cl-letf’.
Warning (bytecomp): ‘easy-menu-add’ is an obsolete function (as of 28.1); use ‘ignore’ instead. [2 times]
Warning (bytecomp): reference to free variable ‘right-toolbar’
Warning (bytecomp): reference to free variable ‘right-toolbar-width’
Warning (bytecomp): reference to free variable ‘left-toolbar’
Warning (bytecomp): reference to free variable ‘left-toolbar-width’
Warning (bytecomp): reference to free variable ‘bottom-toolbar’
Warning (bytecomp): reference to free variable ‘bottom-toolbar-height’
Warning (bytecomp): reference to free variable ‘top-toolbar’
Warning (bytecomp): reference to free variable ‘top-toolbar-height’
Warning (bytecomp): ‘easy-menu-add’ is an obsolete function (as of 28.1); use ‘ignore’ instead.
Warning (bytecomp): ‘ecase’ is an obsolete alias (as of 27.1); use ‘cl-ecase’ instead.
Warning (bytecomp): ‘comint-dynamic-simple-complete’ is an obsolete function (as of 24.1); use ‘completion-in-region’ instead.
Warning (bytecomp): ‘case’ is an obsolete alias (as of 27.1); use ‘cl-case’ instead.
Warning (bytecomp): assignment to free variable ‘toolbar-info-frame’
Warning (bytecomp): the following functions are not known to be defined: set-specifier,
    add-spec-to-specifier, gnuplot-context-completion-at-point
```

**Comment by mtreca** [comment]
Thanks a lot for reporting. My guess is that most of these will be fixed in #59. I will do a check on my local branch.

**Comment by mtreca** [comment]
Should be fixed in #68. Feel free to re-open if I missed anything.
